steppy/base.py

```diff
@@ -180,41 +180,46 @@ def __init__(self,
                  cache_output=False,
                  load_persisted_output=False):
 
-        name = self._format_step_name(name, transformer)
+        self.name = self._format_step_name(name, transformer)
+        if self._name_unique():
+            _ALL_STEPS_NAMES.append(self.name)
+        else:
+            raise ValueError('Step with name "{}", already exist. You must assign unique Step name. '
```

Comment:
@kamil-kaczmarek I think this will cause error when I do the following.

Create pipeline1 with step resize
Create independent pipeline2 with step resize 

This happens when I have train() where I define pipeline and later in evaluate() I also define pipeline.

I think in order to deal with that we may need to drop _ALL_STEPS_NAMES altogether.

Reply:
in this case we will not provide Users with any warning when they create several steps with name `my_model` -> new Step will simply overwrite previous one.